site/blog/tutorials/iasql-on-ec2.mdx

```diff
@@ -0,0 +1,284 @@
+:::
+
+- `instance_type`: this column contains the [instance type](https://aws.amazon.com/ec2/instance-types/) of the EC2 server. We'll use a `t2.small` instance for our deployment.
+- `subnet_id`: this is a foreign key to `subnet` table based on its `id` column. Good thing about IaSQL is that it'll automatically import every subnet in your cloud environment when you install the `iasql_vpc` module. This is not just for the subnets, but all the resources will be imported when installing the module that manages them. You can also manage the infrastructure outside IaSQL afterwards, having IaSQL also import your subsequent changes to the database. We'll use a `SELECT` query from the `subnet` table to fill this column.
+- `tags`: this column is a JSON column. It'll contain all the tags for this instance. We'll use `{"name":"iasql"}` as the value for this column since AWS treats the instance name as a tag.
+- `user_data`: when you launch an instance in Amazon EC2, you have the option of passing user data to the instance that can be used to perform common automated configuration tasks and even run scripts after the instance starts. You can pass two types of user data to Amazon EC2: shell scripts and [cloud-init](https://cloud-init.io/) directives [[1]](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/user-data.html). We'll write the deployment scripts in both of those types to demonstrate, and to have more fun!
+- `key_pair_name`: AWS accepts a key pair for the instance. It can be used to log into the instance through SSH. We'll use the `iasql` key pair we initially created using `key_pair_import` RPC.
+- `region`: we can choose the [AWS region](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Concepts.RegionsAndAvailabilityZones.html#Concepts.RegionsAndAvailabilityZones.Availability) in which the instance will be launched. IaSQL is already multi-region, which means it'll manage your resources in all the regions. So this additional `region` column can be used to determine which region the instance is in, and while `INSERT`ing used to direct IaSQL to create the instance in that region.
+
+##### Bash Startup Script
+
+Alright, we can now write our query to deploy IaSQL on EC2. Let's first write the query with EC2 startup script being a shell script, and then write the query with same functionality using [cloud-init](https://cloud-init.io/) directives.
+
+```sql
+INSERT INTO instance (ami, instance_type, subnet_id, tags, user_data, key_pair_name, region)
+SELECT 'resolve:ssm:/aws/service/canonical/ubuntu/server/22.04/stable/current/amd64/hvm/ebs-gp2/ami-id',
+       't2.small',
+       subnet.id,
+       '{"name":"iasql"}',
+       '#!/bin/bash
+     sudo apt-get upgrade -y
+     curl -fsSL https://get.docker.com -o get-docker.sh
+     sudo sh get-docker.sh
```

Comment:
We can simplify this by using the 100% OSS [`docker.io`](https://packages.ubuntu.com/jammy/docker.io) package instead of Docker's official package.

```suggestion
     sudo apt-get upgrade -y
     sudo apt-get install -y docker.io
```